cpp/src/arrow/compute/kernels/isin-test.cc

```diff
@@ -0,0 +1,410 @@
+                                     {true, false, true, true}, {}, {},
+                                     {false, false, false, false}, {});
+
+  CheckIsIn<StringType, std::string>(
+      &this->ctx_, utf8(), {"test", "", "test2", "test"}, {true, false, true, true},
+      {"test", "", "test2", "test"}, {true, false, true, false}, {true, true, true, true},
+      {});
+
+  // Nulls in left array
+  CheckIsIn<StringType, std::string>(
+      &this->ctx_, utf8(), {"test", "", "test2", "test"}, {false, false, false, false},
+      {"test", "test2", "test"}, {true, true, true}, {false, false, false, false}, {});
+
+  // Nulls in right array
+  CheckIsIn<StringType, std::string>(&this->ctx_, utf8(), {"test", "test2", "test"},
+                                     {true, true, true}, {"test", "", "test2", "test"},
+                                     {false, false, false, false}, {false, false, false},
+                                     {});
+
+  // Both array have Nulls
+  CheckIsIn<StringType, std::string>(
+      &this->ctx_, utf8(), {"test", "", "test2", "test"}, {false, false, false, false},
+      {"test", "", "test2", "test"}, {false, false, false, false},
+      {true, true, true, true}, {});
+
+  // Both array are empty
+  CheckIsIn<StringType, std::string>(&this->ctx_, utf8(), {}, {}, {}, {}, {}, {});
+
+  // Empty left array
+  CheckIsIn<StringType, std::string>(&this->ctx_, utf8(), {}, {},
+                                     {"test", "", "test2", "test"},
+                                     {true, false, true, false}, {}, {});
+
+  // Empty right array
+  CheckIsIn<StringType, std::string>(&this->ctx_, utf8(), {"test", "", "test2", "test"},
+                                     {true, false, true, true}, {}, {},
+                                     {false, false, false, false}, {});
+}
+
+TEST_F(TestIsInKernel, BinaryResizeTable) {
+  const int32_t kTotalValues = 10000;
+#if !defined(ARROW_VALGRIND)
+  const int32_t kRepeats = 10;
+#else
+  // Mitigate Valgrind's slowness
+  const int32_t kRepeats = 3;
+#endif
+
+  std::vector<std::string> values;
+  std::vector<std::string> member_set;
+  std::vector<bool> expected;
+  char buf[20] = "test";
+
+  for (int32_t i = 0; i < kTotalValues * kRepeats; i++) {
+    int32_t index = i % kTotalValues;
+
+    ASSERT_GE(snprintf(buf + 4, sizeof(buf) - 4, "%d", index), 0);
+    values.emplace_back(buf);
+    member_set.emplace_back(buf);
+    expected.push_back(true);
+  }
+
+  CheckIsIn<BinaryType, std::string>(&this->ctx_, binary(), values, {}, member_set, {},
+                                     expected, {});
+
+  CheckIsIn<StringType, std::string>(&this->ctx_, utf8(), values, {}, member_set, {},
+                                     expected, {});
+}
+
+TEST_F(TestIsInKernel, IsInFixedSizeBinary) {
+  CheckIsIn<FixedSizeBinaryType, std::string>(
+      &this->ctx_, fixed_size_binary(5), {"bbbbb", "", "bbbbb", "aaaaa", "ccccc"},
```

Comment:
I'm surprised this works for fixed size binary (I would think we need all values to be equal length, even if one is null)?

Reply:
@emkornfield  are you able to suggest how to fix this?
There is a similar test in hash-test.cc as well.